lib/dialects/mssql/index.js

```diff
@@ -316,7 +316,17 @@ Object.assign(Client_MSSQL.prototype, {
   _setReqInput(req, i, binding) {
     if (typeof binding == 'number') {
       if (binding % 1 !== 0) {
-        req.input(`p${i}`, this.driver.Decimal(38, 10), binding);
+        let precision = 38;
+        let scale = 10;
+        const bindingParts = binding.toString().split('.');
+        if (bindingParts.length === 2) {
+          scale = Math.min(bindingParts[1].length, scale);
+          precision = Math.min(
+            bindingParts[0].length + bindingParts[1].length,
+            precision
+          );
+        }
+        req.input(`p${i}`, this.driver.Decimal(precision, scale), binding);
```

Comment:
How about not trying to write javascript number (double) as a decimal, but as `this.driver.Float(53)` to try to write exact value represented by `number` to the database?

If writing decimal to float type of column in this way works, I don't see why other way around wouldn't work even better (without losing any precicion). 

If one really wants to write decimals to DB I suppose string or some other exact format should be used.

Reply:
Makes me wonder if currently writing floats is also broken here....

Reply:
Do you want me to add more tests?

Reply:
Yes. I added comment about it to test case file.

Reply:
Yes. And to change this implementation to write 

```
req.input(`p${i}`, this.driver.Float(53), binding);
```

instead of `Decimal`.

Reply:
I will give it a try and add more tests